solution1/solution1.go

```diff
@@ -0,0 +1,404 @@
+	flag.Parse()
+
+	return config
+}
+
+func mainError(ctx context.Context, config Config) error {
+	k8sClient, err := newHttpClient(config)
+	if err != nil {
+		return fmt.Errorf("creating K8s client: %s", err)
+	}
+
+	// Create Custom Resource Definition.
+	{
+		log.Printf("creating custom resource")
+
+		// crdJson content in YAML format can be found in crd.yaml file.
+		crdJson := `{
+			"apiVersion": "apiextensions.k8s.io/v1beta1",
+			"kind": "CustomResourceDefinition",
+			"metadata": {
+				"name": "mysqlconfigs.containerconf.de"
+			},
+			"spec": {
+				"group": "containerconf.de",
+				"version": "v1",
+				"scope": "Namespaced",
+				"names": {
+					"plural": "mysqlconfigs",
+					"singular": "mysqlconfig",
+					"kind": "MySQLConfig",
+					"shortNames": []
+				}
+			}
+		}`
+
+		url := config.K8sServer + "/apis/apiextensions.k8s.io/v1beta1/customresourcedefinitions"
+		res, err := k8sClient.Post(url, "application/json", strings.NewReader(crdJson))
+		if err != nil {
+			return fmt.Errorf("creating custom resource: requesting url=%s: %s", url, err)
+		}
+
+		body := readerToBytesTrimSpace(res.Body)
+		res.Body.Close()
+
+		if res.StatusCode == http.StatusOK || res.StatusCode == http.StatusCreated {
+			log.Printf("creating custom resource: created")
+		} else {
+			alreadyExists := false
+			errStr := "bad status"
+
+			// Check if already exists.
+			if res.StatusCode == http.StatusConflict {
+				r, err := isStatusAlreadyExists(body)
+				if err != nil {
+					errStr = err.Error()
+				} else {
+					alreadyExists = r
+				}
+			}
+
+			if alreadyExists {
+				log.Printf("creating custom resource: already exists")
+			} else {
+				return fmt.Errorf("creating custom resource: %s status=%d body=%#q", errStr, res.StatusCode, body)
+			}
+		}
+	}
+
+	// Wait for the Custom Resource to be ready.
+	{
+		attempt := 1
+		maxAttempts := 10
+		checkInterval := time.Millisecond * 200
+
+		for ; ; attempt++ {
+			log.Printf("checking custom resource readiness attempt=%d", attempt)
+
+			url := config.K8sServer + "/apis/apiextensions.k8s.io/v1beta1/customresourcedefinitions"
+			res, err := k8sClient.Get(url)
+			if err != nil {
+				return fmt.Errorf("checking custom resource readiness attempt=%d url=%s: %s", attempt, url, err)
+			}
+
+			body := readerToBytesTrimSpace(res.Body)
+			res.Body.Close()
+
+			if res.StatusCode == http.StatusOK {
+				log.Printf("checking custom resource readiness attempt=%d: ready", attempt)
+				break
+			}
+
+			if attempt == maxAttempts {
+				return fmt.Errorf("checking custom resource readiness attempt=%d: bad status status=%d body=%#q", attempt, res.StatusCode, body)
+			}
+
+			log.Printf("checking custom resource readiness attempt=%d: not ready yet", attempt)
+			time.Sleep(checkInterval)
+		}
+	}
+
+	// Start reconciliation loop.
+	reconciliationCnt := 1
+	reconciliationInterval := time.Second * 1
+	for ; ; reconciliationCnt++ {
```

Comment:
Nitpick but the comment here could say it's an infinite loop and that its intended.